test/tests/services/fields-flattener.test.js

```diff
@@ -0,0 +1,197 @@
+import Interface from 'forest-express';
+import FieldsFlattener from '../../../src/services/fields-flattener';
+
+describe('service > fields-flattener', () => {
+  let errorLoggerSpy;
+  let warnLoggerSpy;
+
+  const generateEngineSchema = () => ({
+    fields: [{
+      field: 'engine',
+      type: {
+        fields: [
+          { field: 'horsepower', type: 'String' },
+          { field: 'cylinder', type: 'Number' },
+          {
+            field: 'identification',
+            type: {
+              fields: [
+                { field: 'manufacturer', type: 'String' },
+                { field: 'serialNumber', type: 'String' },
+              ],
+            },
+          },
+          { field: 'company', type: 'String', reference: 'companies._id' },
+        ],
+      },
+    }],
+  });
+
+
+  beforeAll(() => {
+    errorLoggerSpy = jest
+      .spyOn(Interface.logger, 'error');
+    warnLoggerSpy = jest
+      .spyOn(Interface.logger, 'warn');
+  });
+
+  describe('validating the flatten property', () => {
+    describe('when the flatten property is not an array', () => {
+      it('should display an error message', () => {
+        expect.assertions(1);
+
+        jest.resetAllMocks();
+        const fieldsFlattener = new FieldsFlattener({}, { });
+        fieldsFlattener.validateOptions();
+
+        expect(errorLoggerSpy).toHaveBeenCalledTimes(1);
```

Comment:
would be nice to validate the error messages as well in all the logger tests ;)

Reply:
👍